src/logger.rs

```diff
@@ -0,0 +1,44 @@
+use log::{LevelFilter, Log, Metadata, Record};
+
+/// A simple logger that writes error messages to stderr and other messages to stdout.
+#[derive(Debug)]
+pub struct Logger {
+    pub level: LevelFilter,
+}
+
+impl Logger {
+    pub fn new() -> Self {
+        Self {
+            level: LevelFilter::Info,
+        }
+    }
+
+    pub fn with_level(mut self, level: LevelFilter) -> Self {
```

Comment:
Would it make sense to not make self mutable here? I saw there is an `..` operator. I'm not sure, but it could be used like this?
```
{ 
  level = level,
  ..self
}
```

Reply:
Indeed, I've added a commit to remove the mut. I also added another helper to toggle print to stderr.